activemq-broker/src/main/java/org/apache/activemq/broker/TransactionBroker.java

```diff
@@ -291,13 +294,43 @@ public void send(ProducerBrokerExchange producerExchange, final Message message)
             transaction = getTransaction(context, message.getTransactionId(), false);
         }
         context.setTransaction(transaction);
+
         try {
+            // [AMQ-9344] Limit uncommitted transactions by count
+            verifyUncommittedCount(producerExchange, transaction, message);
```

Comment:
```suggestion
            if (maxUncommittedCount > 0) {
                verifyUncommittedCount(producerExchange, transaction, message);
            }
```
I would move the check here for enabled, it's a small performance hit (likely not really noticable) but still I don't see a reason to always invoke a method for every single message when this is disabled.